app/org/sagebionetworks/bridge/validators/StudyValidator.java

```diff
@@ -59,6 +59,10 @@ public void validate(Object obj, Errors errors) {
                 errors.rejectValue("identifier", "must be at least 2 characters");
             }
         }
+        if (study.getActivityEventKeys().stream()
+                .anyMatch(k -> !k.matches(BridgeConstants.BRIDGE_IDENTIFIER_PATTERN))) {
+            errors.rejectValue("activityEventKey", "must contain only lower-case letters and/or numbers with optional dashes");
+        }
```

Comment:
We changed this method to just prevent deletion of taskIdentifiers and dataGroups, you could do the same with eventKeys if you want.